nuget/spec/dependabot/nuget/update_checker/version_finder_spec.rb

```diff
@@ -288,15 +327,15 @@
 
           custom_v2_nuget_versions_url =
             "https://www.nuget.org/api/v2/FindPackagesById()?id=" \
-            "'Microsoft.Extensions.DependencyModel'"
+            "'#{dependency_name}'"
```

Comment:
I was looking at this test locally and I think part of this was a really old copy/paste bug; there is no `4.8.1` of `Microsoft.Extensions.DependencyModel`, but there _is_ of `Autofac`, so using `VCR=all` might not work.

Reply:
Interesting!

These tests are no longer end-to-end, since I removed VCR from them, because:
* VCR was causing problems in our CI which does not allow `VCR=all` mode.
* There were too many cassettes committed to the repo.

So I changed these to be unit specs for each class, which is what other ecosystems normally do. Because of this, it no longer really matters if the dependency matches a real dependency or not.

For end to end tests we use https://github.com/dependabot/smoke-tests, so it'd be good to move a end to end test there.